src/pages/index.js

```diff
@@ -1,215 +1,275 @@
-    profileJobElement.textContent = userData.about;
-
-    cards.forEach((item) => {
-      const cardElement = getCardElement(item);
-      cardList.append(cardElement);
+closeButtons.forEach((button) => {
+  // Find the closest popup only once
+  const popup = button.closest(".modal");
+  // Set the listener
+  button.addEventListener("click", () => toggleModal(popup));
+});
+
+//Edit Profile Modal Functions/Handlers/Listeners
+function toggleModal(modal) {
+  modal.classList.toggle("modal_is-open");
+
+  if (modal.classList.contains("modal_is-open")) {
+    document.addEventListener("keyup", closeWithEscape);
+  } else {
+    document.removeEventListener("keyup", closeWithEscape);
+  }
+}
+
+const closeWithEscape = (evt) => {
+  if (evt.key === "Escape") {
+    const currModal = page.querySelector(".modal_is-open");
+    toggleModal(currModal);
+  }
+};
+
+function handleProfileFormSubmit(evt) {
+  function makeRequest() {
+    return myApi
+      .editUserInfo({ name: editModalName.value, about: editModalDesc.value })
+      .then((data) => {
+        profileName.textContent = data.name;
+        profileDesc.textContent = data.about;
+        toggleProfileModal();
+        evt.preventDefault();
+      });
+  }
+
+  handleSubmit(makeRequest, evt);
+}
+
+function handleAvatarSubmit(evt) {
+  function makeRequest() {
+    return myApi.editAvatarInfo({ avatar: avatarInput.value }).then((data) => {
+      avatarImage.src = data.avatar;
+      toggleModal(avatarModal);
     });
-  })
-  .catch(console.error);
+  }
+  handleSubmit(makeRequest, evt);
+}
 
-const profileEditButton = document.querySelector(".profile__edit-button");
-const profileEditModal = document.querySelector("#edit-modal");
-const profileCloseButton = profileEditModal.querySelector(
-  ".modal__button-close"
-);
-const profileSubmitButton = profileEditModal.querySelector(
-  ".modal__button-submit"
-);
+function toggleProfileModal() {
+  toggleModal(editModal);
+}
+
+profileEditButton.addEventListener("click", () => {
+  editModalName.value = profileName.textContent;
+  editModalDesc.value = profileDesc.textContent;
+  resetValidation(profileFormElement, [editModalName, editModalDesc], settings);
+  toggleProfileModal();
+});
+
+const profileFormElement = document.forms["edit-profile-form"];
+
+profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+
+//New Post Modal Functions/Handlers/Listeners
+
+newPostButton.addEventListener("click", () => {
+  toggleModal(newPostModal);
+});
 
-const cardTemplate = document.querySelector("#card-template");
-const cardList = document.querySelector(".cards__list");
+function handleNewPostFormSubmit(evt) {
+  function makeRequest() {
+    return myApi
+      .addNewCard({
+        name: newPostCaption.value,
+        link: newPostUrl.value,
+      })
+      .then((card) => {
+        renderCard(card);
+        toggleModal(newPostModal);
+        disableButton(buttonElement, settings);
+        evt.target.reset();
```

Comment:
`evt.target.reset();`  is not needed here because `handleSubmit` contains it already

**Please, correct it everywhere in the code**